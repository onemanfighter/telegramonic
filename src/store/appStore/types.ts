import { StateCreator } from 'zustand';
import {
  AlertStateSlice,
  DrawerStateSlice,
  ModalStateSlice,
  ThemeStateSlice,
} from '../slice';

export interface AppStoreState {
  Theme: ThemeStateSlice;
  Alert: AlertStateSlice;
  Modal: ModalStateSlice;
  Drawer: DrawerStateSlice;
}

export type AppStoreSlice<T> = StateCreator<
  AppStoreState,
  [['zustand/immer', never]],
  [],
  T
>;
