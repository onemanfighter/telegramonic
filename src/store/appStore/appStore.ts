import { create } from 'zustand';

import { immer } from 'zustand/middleware/immer';
import { AppStoreState } from './types';
import { createThemeSlice, createAlertSlice } from '@store/slice';

export const appStore = create<AppStoreState>()(
  immer((...api) => ({
    Theme: createThemeSlice(...api),
    Alert: createAlertSlice(...api),
  })),
);
