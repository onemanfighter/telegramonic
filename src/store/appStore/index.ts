export type { AppStoreState, AppStoreSlice } from './types';
export { appStore } from './appStore';
