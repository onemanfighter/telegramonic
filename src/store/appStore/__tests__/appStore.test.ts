import { renderHook } from '@testing-library/react';
import { appStore } from '../appStore';

describe('appStore', () => {
  it('should have all the slices', () => {
    const { result } = renderHook(() => appStore((state) => state));

    expect(result.current).toMatchSnapshot();
    expect(result.current.Alert).toBeDefined();
    expect(result.current.Drawer).toBeDefined();
    expect(result.current.Modal).toBeDefined();
  });
});
