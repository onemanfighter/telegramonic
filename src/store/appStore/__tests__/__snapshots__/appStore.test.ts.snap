// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appStore should have all the slices 1`] = `
{
  "Alert": {
    "alert": {
      "description": "",
      "showIcon": true,
      "status": undefined,
      "title": "",
      "variant": "subtle",
    },
    "alertVisibleState": false,
    "setAlertStateData": [Function],
    "setAlertVisible": [Function],
  },
  "Drawer": {
    "closeDrawer": [Function],
    "drawerData": {
      "children": null,
      "footer": null,
      "isOpen": false,
      "onModalClose": [Function],
      "title": "",
    },
    "openDrawer": [Function],
    "openState": "CLOSE",
  },
  "Modal": {
    "closeModal": [Function],
    "modalData": {
      "message": "",
      "onModalClose": [Function],
      "onPrimaryButtonClick": [Function],
      "primaryButtonLabel": "",
      "title": "",
    },
    "openModal": [Function],
    "openState": "CLOSE",
  },
}
`;
